tket/src/Gate/Gate.cpp

```diff
@@ -459,6 +459,96 @@ bool Gate::is_clifford() const {
   }
 }
 
+bool Gate::test_exchange_invariance_of_ports(
+    unsigned port1, unsigned port2) const {
+  if (port1 == port2) {
+    // exchanging with self is always symmetric
+    return true;
+  }
+  OpType optype = get_type();
+  switch (optype) {
```

Comment:
Hmmm, is the efficiency penalty for that not a bit high?

Reply:
> Hmmm, is the efficiency penalty for that not a bit high?

Yes it could be, let's leave it as it is for now.

Reply:
I can throw an error as a default so that It can't be forgotten

Reply:
> I can throw an error as a default so that It can't be forgotten

I think actually better not to throw an error, which could break usage, but to let it return `false`, which is at least harmless.

There are already several places in the code that have to be updated whenever a new `OpType` is added; this is a long-standing annoyance but having one more maybe is not so bad, especially if forgetting it doesn't actually break anything.